pyproject.toml

```diff
@@ -72,6 +72,7 @@ gcsfs = { version = ">=2023.1.0,<2024.1.0", optional = true }
 psycopg2-binary = { version = ">=2.9.6", optional = true }
 sqlalchemy = { version = "^2.0.18", optional = true }
 getdaft = { version = ">=0.2.12", optional = true }
+numpy = { version = "^1.22.4", optional = true }
```

Comment:
This is interesting. Do you know if this is because `numpy` 2.0.1 which has been released a couple of hours ago: https://pypi.org/project/numpy/#history

2.0.0 is a bit over a month old?

Reply:
Ah that's a good point. Let me check if we can raise the bar to 2.0.0

Reply:
Using numpy 2.0.0 results in the same error as well. I'm wondering if we just didn't notice it because for some reason, all of our environments were installing numpy `<2.0.0`.

Reply:
Yes, I think that's the case. This is exactly what the `poetry.lock` file is for, and it looks like it still installs `1.24.4`:

https://github.com/apache/iceberg-python/blob/d8bc1ca9af7957ce4d4db99a52c701ac75db7688/poetry.lock#L2538-L2573

Reply:
Pandas doesn't declares an upper bound: https://pandas.pydata.org/docs/getting_started/install.html#required-dependencies

Reply:
Poetry doesn't enforce the constraint right now, I believe we need to add it to the groups that are using `numpy`:
https://github.com/apache/iceberg-python/blob/32d8fe555ff4b92ef4c980232dc0157283ea1013/pyproject.toml#L576C1-L591

Reply:
Got it - thank you for the pointer!

Reply:
Updated: https://github.com/apache/iceberg-python/pull/951/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R580-R583